### ISSUE TITLE
Error when changing Chat Channel colors.

### ISSUE THREAD

**Issue by CrusaderEsper** [Chalsean/XFaction#402]
When opening up the menu to change chat colors and going to select a new color, a pile of errors all get thrown at once.

```
51x ...s/XFaction/Core/System/Network/ChannelCollection.lua:183: attempt to call method 'ObjectName' (a nil value)
[string "@XFaction/Core/System/Network/ChannelCollection.lua"]:183: in function `catchBlock'
[string "@XFaction/Core/Function/Exception/try-catch-finally.lua"]:30: in function `catch'
[string "@XFaction/Core/System/Network/ChannelCollection.lua"]:182: in function `_Function'
[string "@XFaction/Core/System/Wrapper/EventCollection.lua"]:40: in function <...Ons/XFaction/Core/System/Wrapper/EventCollection.lua:22>
[string "=[C]"]: in function `ChangeChatColor'
[string "@Blizzard_ChatFrame/Mainline/ChatConfigFrame.lua"]:1436: in function `swatchFunc'
[string "@Blizzard_FrameXML/Mainline/ColorPickerFrame.lua"]:8: in function <...dOns/Blizzard_FrameXML/Mainline/ColorPickerFrame.lua:4>
[string "=[C]"]: in function `SetColorRGB'
[string "@Blizzard_FrameXML/Mainline/ColorPickerFrame.lua"]:75: in function `SetupColorPickerAndShow'
[string "@Blizzard_ChatFrame/Mainline/ChatConfigFrame.lua"]:1493: in function `MessageTypeColor_OpenColorPicker'
[string "*ChatConfigFrame.xml:170_OnClick"]:1: in function <[string "*ChatConfigFrame.xml:170_OnClick"]:1>

Locals:
err = "...s/XFaction/Core/System/Network/ChannelCollection.lua:178: attempt to call method 'ObjectName' (a nil value)"
(*temporary) = <function> defined @XFaction/Core/Function/Logger/Logger.lua:27
(*temporary) = <table> {
 Guilds = <table> {
 }
 Cache = <table> {
 }
 Targets = <table> {
 }
 Function = <table> {
 }
 Factories = <table> {
 }
 CoreInit = <function> defined @XFaction/Core/Initialize.lua:6
 DecodeBNetMessage = <function> defined @XFaction/Core/System/Network/Decode.lua:135
 Lib = <table> {
 }
 Expansions = <table> {
 }
 Friends = <table> {
 }
 AddonName = "XFaction"
 DeserializeUnitData = <function> defined @XFaction/Core/System/Network/Decode.lua:53
 Start = 1716014640
 Enum = <table> {
 }
 SingleLine = <function> defined @XFaction/Core/Function/Logger/Logger.lua:72
 SetupRealms = <function> defined @XFaction/Core/UI/Configuration/General.lua:48
 DecodeChatMessage = <function> defined @XFaction/Core/System/Network/Decode.lua:130
 EncodeBNetMessage = <function> defined @XFaction/Core/System/Network/Encode.lua:81
 Regions = <table> {
 }
 Trace = <function> defined @XFaction/Core/Function/Logger/Logger.lua:51
 DataDumper = <function> defined @XFaction/Core/Function/Logger/Logger.lua:68
 ChangeLog = <table> {
 }
 Addons = <table> {
 }
 Versions = <table> {
 }
 Icons = <table> {
 }
 Version = <table> {
 }
 Confederate = <table> {
 }
 Mailbox = <table> {
 }
 Timers = <table> {
 }
 SetupMenus = <function> defined @XFaction/Core/UI/Configuration/General.lua:112
 Media = <table> {
 }
 Settings = <table> {
 }
 Realms = <table> {
 }
 Initialized = true
 Links = <table> {
 }
 Config = <table> {
 }
 Object = <table> {
 }
 InitProfile = <function> defined @XFaction/Core/UI/Configuration/General.lua:1018
 Error = <function> defined @XFaction/Core/Function/Logger/Logger.lua:27
 ConfigDB = <table> {
 }
 DoubleLine = <function> defined @XFaction/Core/Function/Logger/Logger.lua:76
 Defaults = <table> {
 }
 Metrics = <table> {
 }
 Warn = <function> defined @XFaction/Core/Function/Logger/Logger.lua:35
 ConfigInitialize = <function> defined @XFaction/Core/UI/Configuration/General.lua:936
 Hooks = <table> {
 }
 Zones = <table> {
 }
 Install = <function> defined @XFaction/Core/UI/Configuration/General.lua:1032
 GetLinkKey = <function> defined @XFaction/Core/System/Network/Link.lua:52
 WoW = <table> {
 }
 Continents = <table> {
 }
 Frames = <table> {
 }
 RGBPercToHex = <function> defined @XFaction/Core/Function/Font/RGBToHex.lua:4
 Nodes = <table> {
 }
 Verbosity = 0
 Events = <table> {
 }
 ObjectsEquals = <function> defined @XFaction/Core/System/Abstract/Object.lua:142
 EncodeChatMessage = <function> defined @XFaction/Core/System/Network/Encode.lua:73
 Debug = <function> defined @XFaction/Core/Function/Logger/Logger.lua:47
 Teams = <table> {
 }
 Name = "XFaction"
 Stop = <function> defined @XFaction/Core/Initialize.lua:94
 Class = <table> {
 }
 SortGuildColumns = <function> defined @XFaction/Core/UI/Configuration/DataText.lua:21
 Handlers = <table> {
 }
 Player = <table> {
 }
 Info = <function> defined @XFaction/Core/Function/Logger/Logger.lua:43
 SerializeUnitData = <function> defined @XFaction/Core/System/Network/Encode.lua:38
 Professions = <table> {
 }
 Options = <table> {
 }
 DataText = <table> {
 }
 Title = "|cffFF4700X|r|cff33ccffFaction|r"
}
(*temporary) = nil
(*temporary) = Frame {
 0 = <userdata>
}
(*temporary) = "attempt to call method 'ObjectName' (a nil value)"
XF = <table> {
 Guilds = <table> {
 }
 Cache = <table> {
 }
 Targets = <table> {
 }
 Function = <table> {
 }
 Factories = <table> {
 }
 CoreInit = <function> defined @XFaction/Core/Initialize.lua:6
 DecodeBNet
```

**Comment by Chalsean** [comment]
im sorry, need to do a better job of staying on top of my emails.  i will take a look this weekend and get a fix out

**Comment by Chalsean** [comment]
Should be fixed in 4.12.2 release, please let me know if you continue to encounter issues.
